core/src/main/java/io/questdb/std/GenericLexer.java

```diff
@@ -257,15 +257,23 @@ public CharSequence next() {
                 case '\'':
                     if (c == '\'') {
                         _hi += 2;
-                        return last = flyweightSequence;
+                        if (hasNext() && content.charAt(_pos) == '\'') {
```

Comment:
`hasNext()` call is quite heavy in this context, it can be replaced with `_pos < _len`. In fact `while(hasNext())` is also too heavy